__tests__/app.test.js

```diff
@@ -167,3 +167,83 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: inserts a new comment and returns the inserted comment to the client ", () => {
+    const newComment = { username: "butter_bridge", body: "cool stuff" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toEqual({
+          comment_id: 19,
+          article_id: 1,
+          votes: 0,
+          created_at: expect.any(String),
+          author: "butter_bridge",
+          body: "cool stuff",
+        });
+      });
+  });
+  test("400: returns an error message when passed an invalid id", () => {
+    const newComment = { username: "butter_bridge", body: "cool stuff" };
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("404: returns an error message when passed a valid but non-existent article id", () => {
+    const newComment = { username: "butter_bridge", body: "cool stuff" };
+    return request(app)
+      .post("/api/articles/500/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Article does not exist");
+      });
+  });
+  test("400: returns an error message when passed an invalid body with missing required field", () => {
+    const newComment = { body: "cool stuff" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("400: returns an error message when passed an invalid body with user that doesn't exist", () => {
+    const newComment = { username: "coolcat", body: "cool stuff" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
```

Comment:
since this is an error due to "coolcat" (sadly) not existing, is there a more appropriate error code?

Reply:
I would say a 404 would be a bit more appropriate for this since the error is that the user doen't exist

Reply:
have resolved this and Carrie said via slack to merge once this is changed :)